logic/resolvable/Concludable.java

```diff
@@ -334,24 +407,73 @@ public Relation asRelation() {
 
         @Override
         AlphaEquivalence alphaEquals(Relation that) {
-            return constraint().alphaEquals(that.constraint());
+            return relation().alphaEquals(that.relation()).validIfAlphaEqual(isa, that.isa)
+                    .validIfAlphaEqual(labels, that.labels);
         }
     }
 
-    public static class Has extends Concludable<HasConstraint> {
+    /**
+     * `Has` handles these concludable patterns:
+     * `{ $x has $a; }`,
+     * `{ $x has age $a; }`,
+     * `{ $x has age 30; }`,
+     * `{ $x has $a; $a isa age; }`,
+     * `{ $x has $a; $a 30 isa age; }`,
+     * `{ $x has $a; $a 30; }`
+     */
+    public static class Has extends Concludable {
+
+        private final HasConstraint has;
+        private final IsaConstraint isa;
+        private final Set<ValueConstraint<?>> values;
+
+        private Has(Conjunction conjunction, HasConstraint has, @Nullable IsaConstraint isa, Set<ValueConstraint<?>> values) {
+            super(conjunction);
+            this.has = has;
+            this.isa = isa;
+            this.values = values;
+        }
+
+        public static Has of(HasConstraint has, @Nullable IsaConstraint isa, Set<ValueConstraint<?>> values, Set<LabelConstraint> labels) {
+            Conjunction.Cloner cloner;
+            IsaConstraint clonedIsa;
+            if (isa == null) {
+                cloner = Conjunction.Cloner.cloneExactly(values, has);
+                clonedIsa = null;
+            } else {
+                cloner = Conjunction.Cloner.cloneExactly(labels, values, isa, has);
+                clonedIsa = cloner.getClone(isa).asThing().asIsa();
+            }
+            return new Has(cloner.conjunction(), cloner.getClone(has).asThing().asHas(), clonedIsa,
+                           new HashSet<>(Iterators.iterate(values).map(cloner::getClone).map(c -> c.asThing().asValue()).toSet()));
+        }
 
-        public Has(final HasConstraint constraint, @Nullable IsaConstraint isaConstraint, Set<ValueConstraint<?>> valueConstraints) {
-            super(ConstraintCopier.copyConstraint(constraint), core(isaConstraint, valueConstraints));
+        public HasConstraint has() {
+            return has;
         }
 
         @Override
+        public Set<Constraint> concludableConstraints() {
+            Set<Constraint> constraints = new HashSet<>();
+            constraints.add(has);
+            if (isa != null) constraints.add(isa);
+            constraints.addAll(equalsConstraints(values));
+            return constraints;
+        }
+
+        @Override
+        ResourceIterator<Unifier> unify(Rule.Conclusion conclusion, ConceptManager conceptMgr) {
+            if (conclusion.isHas()) return unify(conclusion.asHas(), conceptMgr);
```

Comment:
If it's an explicit has or variable has, both are a has, so cast to that and unify.